build.rs

```diff
@@ -85,24 +85,42 @@ fn zip_template_and_build_dylint_driver(manifest_dir: PathBuf, out_dir: PathBuf)
     // > * Any sub-packages will be skipped (any subdirectory that contains a Cargo.toml file).
     //
     // (from https://doc.rust-lang.org/cargo/reference/manifest.html#the-exclude-and-include-fields)
-    std::fs::rename(
-        ink_dylint_driver_dir.join("_Cargo.toml"),
-        ink_dylint_driver_dir.join("Cargo.toml"),
-    )?;
-
-    let res = build_and_zip_dylint_driver(
-        ink_dylint_driver_dir.clone(),
-        out_dir,
-        dylint_driver_dst_file,
-    );
+    let original_name = ink_dylint_driver_dir.join("_Cargo.toml");
+    let original_name = original_name.canonicalize().map_err(|err| {
```

Comment:
I've also added path-canonicalization here now.

Reply:
Why not just canonicalize `ink_dylint_driver_dir` in the first place, then both `original_name` and `tmp_name` will be canonical.